imagetest/test_suites/oslogin/ssh_test.go

```diff
@@ -271,3 +276,192 @@ func TestAdminSSH(t *testing.T) {
 		t.Fatalf("sudoers file does not contain user or necessary configurations")
 	}
 }
+
+func Test2FASSH(t *testing.T) {
+	ctx := utils.Context(t)
+
+	// Secret manager client.
+	secretClient, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to create secret manager client: %v", err)
+	}
+	defer secretClient.Close()
+
+	user, err := utils.AccessSecret(ctx, secretClient, normal2FAUser)
+	if err != nil {
+		t.Fatalf("failed to get user info: %v", err)
+	}
+	posix := getPosix(user)
+
+	privateSSHKey, err := utils.AccessSecret(ctx, secretClient, normal2FASSHKey)
+	if err != nil {
+		t.Fatalf("failed to get user key: %v", err)
+	}
+
+	// Manually set up SSH.
+	vmname, err := utils.GetRealVMName("twofa")
+	if err != nil {
+		t.Fatalf("failed to get hostname: %v", err)
+	}
+	hostname := fmt.Sprintf("%s:22", vmname)
+	signer, err := ssh.ParsePrivateKey([]byte(privateSSHKey))
+	if err != nil {
+		t.Fatalf("failed to parse private key: %v", err)
+	}
+
+	cb := func(name, instruction string, questions []string, echos []bool) (answers []string, err error) {
+		if len(questions) == 0 {
+			return
+		}
+
+		answers = make([]string, 1)
+		firstQuestionRegex, err := regexp.Compile(".*Enter.*number.*")
+		if err != nil {
+			return
+		}
+		if firstQuestionRegex.MatchString(questions[0]) {
+			// 1 is the response for Authenticator OTP
+			answers[0] = "1"
+			return
+		}
+
+		// If not the first question, input code for two-factor.
+		ctx := context.Background()
```

Comment:
Why are we creating a new context here? if you need a new context please derive the main one, otherwise just use the main one.